connect to neo4j getting started guide (#281)
neo4j/cypher-builder: docs/modules/ROOT/pages/getting-started/connecting-to-neo4j.adoc

[[connecting-to-neo4j]]

> :description: This tutorial demonstrates how to connect to Neo4j and utilize Cypher Builder.

Reviewer: @lidiazuin domain, but to me, from the sentence seems the two objectives of the sentence are disjoined "Connect to Neo4j" and "Use Cypher Builder". I would go more with something like:
"This tutorial demonstrates how to execute your Cypher Builder queries in a Neo4j instance using the Neo4j Javascript driver".

Reviewer 2: I guess we can go for something simpler, if correct:

```suggestion
:description: This tutorial demonstrates how to use the Cypher Builder to connect to Neo4j.
```

Reviewer: In this way, it will make the sentence incorrect as you don't use Cypher Builder to connect to Neo4j, the component that "connects" is the Neo4j Javascript driver.

Reviewer 2: aha, so your suggestion makes more sense!

Author: ```suggestion
:description: "This tutorial demonstrates how to execute your Cypher Builder queries in a Neo4j instance using the Neo4j Javascript driver.
```

Reviewer 2: Without the quotation marks in the beginning :)

----

== Initialize neo4j-driver
In your JavaScript file, initialize the driver following the instructions in the link:https://neo4j.com/docs/javascript-manual/current/[JavaScript Driver Manual]:

[source, javascript]
----
import Cypher from "@neo4j/cypher-builder";
import neo4j from "neo4j-driver";

const driver = neo4j.driver("neo4j://localhost", neo4j.auth.basic("neo4j", "password"));
----

== Construct your Cypher query

Compose your desired query using the Cypher object. For instance:

[source, javascript]
----
const movie = new Cypher.Node({ labels: ["Movie"] });
const query = new Cypher.Match(movie).return([movie.property("title"), "title"]);
----

This query will translate to:

[source, cypher]
----
MATCH (this0:Movie)
RETURN this0.title AS title
----

== Execute the query

Use `.build` on the query object to generate the Cypher query string and the necessary parameters to pass to `neo4j-driver`

[source, javascript]
----
const { cypher, params } = query.build();
const { records } = await driver.executeQuery(cypher, params);
----

Close the driver connection:

[source, javascript]
----
await driver.close();
----

== Consume the results

> The `records` object can be consumed as explained in the link:https://neo4j.com/docs/javascript-manual/current/[JavaScript Driver Manual]:

Reviewer 2: ```suggestion
The `records` object can be consumed as explained in the link:https://neo4j.com/docs/javascript-manual/current/transactions/[JavaScript Driver documentation]:
```

[source, javascript]
----
for (const record of records) {
    console.log(record.get("title"));
}
----

> With the example data above, this should output:

Reviewer 2: If we change the previous title to "Initialize the driver", then the link should be like that:

```suggestion
With the xref:getting-started/connecting-to-neo4j.adoc#_initialize_the_driver[previous example data], this should output:
```

[source]
----
The Matrix
The Terminal
----

== Conclusion

> After following the steps described, your script should resemble the following:

Reviewer 2: ```suggestion
After following the steps here described, your script should look like this:
```
